nipoppy_new/nipoppy/workflows/dataset_init.py

```diff
@@ -0,0 +1,69 @@
+"""Workflow for init command."""
+
+import logging
+from pathlib import Path
+from typing import Optional
+
+from nipoppy.utils import FPATH_SAMPLE_CONFIG, FPATH_SAMPLE_MANIFEST
+from nipoppy.workflows.base import BaseWorkflow
+
+
+class InitWorkflow(BaseWorkflow):
+    """Workflow for init command."""
+
+    # do not validate since the dataset has not been created yet
+    validate_layout = False
+
+    def __init__(
+        self,
+        dpath_root: Path,
+        fpath_layout: Optional[Path] = None,
+        logger: Optional[logging.Logger] = None,
+        dry_run: bool = False,
+    ):
+        """Initialize the workflow."""
+        super().__init__(
+            dpath_root=dpath_root,
+            name="init",
+            fpath_layout=fpath_layout,
+            logger=logger,
+            dry_run=dry_run,
+        )
+        self.fname_readme = "README.md"
+
+    def run_main(self):
+        """Create dataset directory structure."""
+        # dataset must not already exist
+        if self.dpath_root.exists():
+            raise FileExistsError("Dataset directory already exists")
+
+        # create directories
+        for dpath in self.layout.dpaths:
+            self.mkdir(dpath)
+
+        for dpath, description in self.layout.dpath_descriptions:
+            fpath_readme = dpath / self.fname_readme
+            if description is not None and not self.dry_run:
+                fpath_readme.write_text(f"{description}\n")
+
+        self.logger.info(f"Created an empty dataset at {self.dpath_root}")
+
+        # copy sample config and manifest files
+        self.copy(FPATH_SAMPLE_CONFIG, self.layout.fpath_config)
+        self.copy(FPATH_SAMPLE_MANIFEST, self.layout.fpath_manifest)
+
+        # inform user to edit the sample files
+        self.logger.warning(
```

Comment:
I think we just wanted to remove the "copied this sample file to that path in the new dataset" logs. These would be important because it tells users where the sample files were copied and reminds them that they need to be updated